src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1023,17 +1023,23 @@ public EnvVars getEnvironment() {
         if (buildData.getBuildsByBranchName().size() >= 100) {
             log.println("JENKINS-19022: warning: possible memory leak due to Git plugin usage; see: https://wiki.jenkins-ci.org/display/JENKINS/Remove+Git+Plugin+BuildsByBranch+BuildData");
         }
+        boolean disableAutomaticScheduling = false;
```

Comment:
Please choose a different name, rather than automatic scheduling.  This switch will allow the user to configure a job which will:

* match multiple branches (wild card branch name specification), and
* detect changes on those multiple branches (may happen within the same poll), and
* build only one of those changes (rather than building all of the detected changes)

Isn't it better to tell the user that if they only want to match changes from one branch, they should set their branch specification to only select a single branch?